### ISSUE TITLE
Rails/HttpStatus not working after requiring rubocop-rspec

### ISSUE THREAD

**Issue by a14m** [rubocop-hq/rubocop-rspec#611]
After adding `require: rubocop-rspec` to the `.rubocop.yml`, the `Rails/HttpStatus` doesn't check the code, only the specs 

first reported here: https://github.com/bbatsov/rubocop/issues/5738#issuecomment-388838142

`rubocop-rspec 1.25.1`

**Comment by anthony-robin** [comment]
I think there is a conflict of cops name between `rubocop` and `rubocop-rspec` as `Rails/HttpStatus` is defined in both gems with the same namespace.

I'm not sure how we can fix that 🤔

**Comment by flyerhzm** [comment]
could we rename `Rails/HttpStatus` to `RSpec/Rails/HttpStatus`?

**Comment by dmytro-savochkin** [comment]
Since this is still a problem and it's hard to find out how to fix this maybe it would be better to rename `RSpec/Rails/HttpStatus` to `RSpec/StatusHttp`? Yes, it might be ugly but at least this won't have any conflicts...

**Comment by ixti** [comment]
Any news/plans on fixing this?

**Comment by ixti** [comment]
Right now I have to switch orders of requires to check both cops worked...

**Comment by ixti** [comment]
I have fixed this issue locally with following hacks:

``` ruby
# file: rubocop/fixes/http_status.rb
# frozen_string_literal: true

require "rubocop/cop/rails/http_status"
require "rubocop/cop/rspec/rails/http_status"

RuboCop::Cop::Cop.registry.instance_eval do
  [
    RuboCop::Cop::Rails::HttpStatus,
    RuboCop::Cop::RSpec::Rails::HttpStatus
  ].each do |cop|
    @registry.delete cop.badge
    @departments[cop.department].delete cop
    @cops_by_cop_name[cop.cop_name].delete cop
  end
end

module RuboCop
  module Cop
    module RSpec
      module Rails
        def HttpStatus.badge
          Badge.new("RSpec", "Rails/HttpStatus")
        end
      end
    end
  end
end

[RuboCop::Cop::Rails::HttpStatus, RuboCop::Cop::RSpec::Rails::HttpStatus]
  .each { |cop| RuboCop::Cop::Cop.registry.enlist cop }
```

---

``` yaml
# file: rubocop/fixes/http_status.yml

RSpec/Rails/HttpStatus:
  Description: Enforces use of symbolic or numeric value to describe HTTP status.
  Enabled: true
  EnforcedStyle: symbolic
  SupportedStyles:
  - numeric
  - symbolic
  StyleGuide: http://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/Rails/HttpStatus
```

---

``` yaml
# file: .rubocop.yml

require:
  - ./rubocop/fixes/http_status.rb

inherit:
  - ./rubocop/fixes/http_status.yml
```

**Comment by ixti** [comment]
I've added initial work to fix this issue:

https://github.com/ixti/rubocop-rspec/commit/ac8f0480b2c60a28f501f4a6e50a7580060675db

**Comment by pirj** [comment]
@ixti [the code required to support sub-departments](https://github.com/rubocop-hq/rubocop-rspec/compare/master...ixti:fix/rails-cop-names) looks non-trivial, I'm not sure if it will have a warm welcome in RuboCop.

WDYT of renaming this cop to `RSpecRails/HttpStatus`? `have_http_status` is [an RSpec-Rails' matcher](https://github.com/rspec/rspec-rails/blob/a60adc94694177ca8cf3d126f6684c579e83307e/lib/rspec/rails/matchers/have_http_status.rb)
This naming seems to be more consistent with other departments (`Capybara`, `FactoryBot`).

**Comment by ixti** [comment]
@pirj Yeah, I thought that it won't be welcomed :D It was more like a "what about" example to have some input ideas on. And I totally love your renaming idea.

**Comment by ixti** [comment]
@pirj opened a PR with renaming (#834).

**Comment by haines** [comment]
Did someone end up opening an issue to fix this on the RuboCop side (https://github.com/rubocop-hq/rubocop-rspec/pull/834#issuecomment-570497415)? I can't find one.

**Comment by pirj** [comment]
@haines Unfortunately not. Would you like to handle this?
References:
https://github.com/rubocop-hq/rubocop-rspec/pull/834#issuecomment-570497415
https://github.com/rubocop-hq/rubocop/issues/5251

**Comment by haines** [comment]
Opened https://github.com/rubocop-hq/rubocop/issues/8044

**Comment by pirj** [comment]
Thanks @haines !

**Comment by pirj** [comment]
This should be fixed in #1019 that is part of the release 2.0.
